doc/Makefile

```diff
@@ -7,6 +7,8 @@ html: apidoc
 
 apidoc:
 	$(SPHINXAPIDOC) -F -o _build ../src/catkin_pkg
+	sed -i "/extensions =\[/a 'sphinx_rtd_theme'," _build/conf.py
+	sed -i "s|html_theme =.*$$|html_theme = 'sphinx_rtd_theme'|" _build/conf.py
```

Comment:
Hmm, this works but seems pretty fragile. If the `conf.py` that is generated changes it might break this, no?

Two other options are:

- the template dir option https://www.sphinx-doc.org/en/master/man/sphinx-apidoc.html
- make a `conf.py` and add `sphinx-autodoc` as an extension, e.g. https://eikonomega.medium.com/getting-started-with-sphinx-autodoc-part-1-2cebbbca5365

The benefit of using it how it is now is that you don't need the (mostly) boilerplate conf file and index file. But now we're trying to get that benefit while also changing something about the conf, which I think is tenuous.

That all being said, I'm sure this works right now.

What do you think?

Reply:
Sorry, I didn't get back sooner.

To be clear, I'm fine with the current approach, we just might have to do something with it later, but that's ok.

I think the idea of the templates directory option is that you provide templates for things like the `conf.py` which `sphinx-apidoc` will use instead of the standard boilerplate. I think it's completely separate from the idea of a theme or templates used to say generate html headers and footers. It might be equally fragile thought, to be honest.

> Is the creation of a conf.py and index the only difference between sphinx-apidoc and sphinx with the autodoc extension?

Indeed, if you look at what `sphinx-apidoc` does, it just generates a `conf.py` and `index.rst` and runs `sphinx-build` over it (more or less).

The link I provided for this option basically walks you through creating your own `conf.py` and `index.rst` to accomplish this yourself.